Slime feed buckling

phil235 opened the pull request:
- Slimes now uses buckling to feed and attach themselves on mobs. This removes the silly check in mob/living/Move(), reducing lag.
- Fixes mobs being able to attach slimes onto other mobs.
- Fixes being able to get inside a mech sleeper while buckled (if you buckled during the do_after())

phil235 committed:
Slimes now uses buckling to feed and attach themselves on mobs. This removes the silly check in mob/Move(), reducing lag.
Fixes mobs being able to attach slimes onto other mobs.

phil235 committed:
derp, removing comments, adding a missing Feedstop()

phil235 commented on code/modules/mob/living/living.dm:
```diff
@@ -577,9 +577,6 @@ Sorry Giacom. Please don't be mad :(
 
-	for(var/mob/living/simple_animal/slime/M in oview(1,src))
-		M.UpdateFeed(src)
-
```
the whole reason of this PR.

CosmicScientist commented:
:+1:

Jalleo commented:
:moneybag:

phil235 committed:
derp, let's not make janicart unbuckling runtime.
Also renaming obj/effect/spacevine/buckle_mob() to entangle_mob() since it's not really the standard buckle_mob() and could cause runtimes.

Aranclanos commented:
Sleepers, cryo tubes, mechas, oddyseus sleeper, admin teleports (get and send to), ethereal jaunt and blink won't break the slime buckling and won't take the slime with you. At the moment you move, the slime will teleport on top of you.
Both mecha climbing and mecha sleeper reject it if there's a slime on top of the mob, but only before the cooldown starts. If the slime attaches after the cooldown started, the bug can happen.

the spell and paper that do the teleportation will take the slime with you, they seem to work fine.
The teleporter leaves the slime behind and the hand teleporter also breaks the buckling, but takes the slime with you because of how the blue portal works.

I didn't test with the away mission teleporter.

Razharas commented:
@phil235

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SlimeFeedBuckling

Conflicts:
	code/game/objects/structures/janicart.dm

phil235 committed:
Fixes some cases with mobs attached a mob. Can't enter a mech if attachment is done mid-climb. Can't get inside mech sleeper or closet if you have a mob attached to you. All sorts of teleports now properly teleport the attached mob as well.

Changed the warning message when climbing mech with a mob attached to be more general instead of specifically about slimes.

Getting inside a closet or machine now immediately breaks your pull and the pull of your puller.
Using ethereal jaunt now also immediately break your pull.

Fixes being able to get inside a mech sleeper while buckled by buckling mid-insertion.

phil235 commented:
Alright, it should be good. I dealt with all problematic cases mentioned by aran.
